pkg/metabase/client.go

```diff
@@ -486,14 +497,16 @@ func (c *Client) SetCollectionAccess(ctx context.Context, groupID, collectionID
 	return c.request(ctx, http.MethodPut, "/collection/graph", cPermissions, nil)
 }
 
-func (c *Client) CreateCollectionWithAccess(ctx context.Context, groupID int, name string) (int, error) {
+func (c *Client) CreateCollectionWithAccess(ctx context.Context, groupIDs []int, name string) (int, error) {
 	cid, err := c.CreateCollection(ctx, name)
 	if err != nil {
 		return 0, err
 	}
 
-	if err := c.SetCollectionAccess(ctx, groupID, cid); err != nil {
-		return cid, err
+	for _, gID := range groupIDs {
+		if err := c.SetCollectionAccess(ctx, gID, cid); err != nil {
```

Comment:
Is a group added in permission graph as soon as it is created?
SetCollectionAccess() assumes the gID must exist in the permission graph. If the above is true the code should be ok

Reply:
We could always fetch the permission graph after adding the group and check whether the group exists, but this has not been an issue thus far.